File: pkg/server/pokete/msg/error/invalid_poke.go
package error

import (
	"fmt"
	"github.com/lxgr-linux/pokete/bs_rpc/msg"
)

const InvalidPokeType msg.Type = "pokete.error.invalid_poke"

type InvalidPoke struct {
	msg.BaseMsg
	Error string `json:"error"`
}

func (p InvalidPoke) GetType() msg.Type {
	return InvalidPokeType
}

func NewInvalidPoke(err error) InvalidPoke {
	return InvalidPoke{msg.BaseMsg{}, fmt.Sprintf("%s", err)}
}

File: src/pokete/classes/multiplayer/msg/error/invalid_poke.py
from typing import TypedDict

import pokete.bs_rpc as bs_rpc

INVALID_POKE_TYPE = "pokete.error.invalid_poke"


class InvalidPokeData(TypedDict):
    error: str


class InvalidPoke(bs_rpc.Body):
    def __init__(self, data: InvalidPokeData):
        super().__init__(INVALID_POKE_TYPE, data)
